### ISSUE TITLE
runMATLABCommand step has timeout of 5 minutes

### ISSUE THREAD

**Issue by geerligs** [mathworks/jenkins-matlab-plugin#121]
Hello,

When using the runMATLABCommand step I experience two issues:
- the runMATLABCommand step seems to have a timeout of 5 minutes.
- when having a timeout on the runMATLABCommand step the job is not killed automatically. I need to perform a "hard kill"on Jenkins

Can you confirm these observations? I am running version 2.1.0 of the plugin.

### Timeout of 5 minutes on runMATLABCommand step

To test this I have run the pipeline below. This will set the timeout to 10 minutes and runs a Matlab script which will pause for 6 minutes. 
```
// Declarative Pipeline
pipeline {
    agent any
    options { timestamps() }
    stages {
        stage('test timing 2015'){
            environment {
        			PATH = "C:\\Program Files\\MATLAB\\R2015b\\bin;${PATH}"   // Windows agent
        	}
        	options {
                timeout(time: 10, unit: "MINUTES")
        	}
            steps {
                runMATLABCommand "pause(6*60); ver;pwd"
            } 
        }
    }
}
```
I expect the job to run for 6 minutes and print `ver` and `pwd`. However the job times out after 5 minutes. The Jenkins console shows:
```
11:41:37  Timeout set to expire in 10 min
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] runMATLABCommand
11:41:37  Generating MATLAB script with content:
11:41:37  pause(6*60); ver;pwd
11:41:37  
11:41:37  #################### Starting command output ####################
11:41:37  [d5354205-f476-4c03-b2ff-ac097c28ad0a] $ cmd.exe /C C:\Users\han\AppData\Local\Temp\\d5354205-f476-4c03-b2ff-ac097c28ad0a\run_matlab_command.bat "command_f0d95434_4742_42af_a7a3_76ed0ab5286f"
11:41:39   
11:41:39  To get started, type one of these: helpwin, helpdesk, or demo.
11:41:39  For product information, visit www.mathworks.com.
11:41:39   
11:41:40  Warning: Unable to locate a personal folder for $documents\MATLAB
11:41:40  Warning: Unable to locate a personal folder for $documents\MATLAB
11:46:37  ^CTerminate batch job (Y/N)? 
11:46:38  null
```
When replacing the runMATLABCommand step aby a bat-command the job will run for the requested 6 minutes:
`    bat "START /WAIT TIMEOUT /T 360 /NOBREAK >NUL"`

### Hard kill of the runMATLABCommand step

When decreasing the value for timeout in the pipeline to 1 minute the bat-job will stop after  1 minute:
```
11:52:42  Timeout set to expire in 1 min 0 sec
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] bat
11:52:43  
11:52:43  C:\Users\han\AppData\Local\Jenkins.jenkins\workspace\testen\testMatlab2007b@2>START /WAIT TIMEOUT /T 360 /NOBREAK  1>NUL 
11:53:42  Cancelling nested steps due to timeout
11:53:42  Sending interrupt signal to process
```

However I need to apply a 'hard kill' to kill the Matlab process. Please look at the attached console output:
```
11:48:14  Timeout set to expire in 1 min 0 sec
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] runMATLABCommand
11:48:14  Generating MATLAB script with content:
11:48:14  pause(6*60); ver;pwd
11:48:14  
11:48:14  #################### Starting command output ####################
11:48:14  [f66f0e2d-0297-44a8-86c9-c032eb6a395c] $ cmd.exe /C C:\Users\han\AppData\Local\Temp\\f66f0e2d-0297-44a8-86c9-c032eb6a395c\run_matlab_command.bat "command_6ee4aa6b_021f_4894_9c8f_1641da3c0c58"
11:48:16   
11:48:16  To get started, type one of these: helpwin, helpdesk, or demo.
11:48:16  For product information, visit www.mathworks.com.
11:48:16   
11:48:17  Warning: Unable to locate a personal folder for $documents\MATLAB
11:48:17  Warning: Unable to locate a personal folder for $documents\MATLAB
11:49:14  Cancelling nested steps due to timeout
11:50:14  Body did not finish within grace period; terminating with extreme prejudice
Aborted by Administrator
Aborted by Administrator
Click here to forcibly terminate running steps
Click here to forcibly terminate running steps
Aborted by Administrator
Click here to forcibly kill entire build
Hard kill!
Finished: ABORTED
```

kind regards, Han

**Comment by geerligs** [comment]
I just checked the observed behaviour with matlab-jenkins plugin 2.1.1. It is still the same.

**Comment by nbhoski** [comment]
Thanks @geerligs  i could reproduce it and. i am looking into this.

**Comment by geerligs** [comment]
Hi Nikhil,

good to hear you could reproduce :-) I also experienced this behaviour 
when using the runMATLABTests step. I found this when my unit tests were 
extended.

regards, Han

Nikhil Bhoski schreef op 2020-07-24 08:45:

**Comment by nbhoski** [comment]
Thats correct. This will be reproducible for both steps i have figured out the reason and working on it .

**Comment by nbhoski** [comment]
@geerligs  Fix is released in version 2.1.2

**Comment by geerligs** [comment]
Hi Nikhil, I have just checked and the timeout of 5 mins has gone. Thanks for the fix!
